python/GafferTest/NameValuePlugTest.py

```diff
@@ -0,0 +1,236 @@
+##########################################################################
+#
+#  Copyright (c) 2019, Image Engine Design Inc. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import unittest
+import imath
+
+import IECore
+
+import Gaffer
+import GafferTest
+
+class NameValuePlugTest( GafferTest.TestCase ) :
+
+	def assertPlugSerialises( self, plug ):
+		s = Gaffer.ScriptNode()
+
+		s["n"] = Gaffer.Node()
+		s["n"]["p"] = plug
+
+		s2 = Gaffer.ScriptNode()
+		s2.execute( s.serialise() )
+
+		self.assertEqual( s2["n"]["p"].getName(), plug.getName() )
+		self.assertEqual( s2["n"]["p"].direction(), plug.direction() )
+		self.assertEqual( s2["n"]["p"].getFlags(), plug.getFlags() )
+
+		self.assertEqual( s2["n"]["p"].keys(), plug.keys() )
+		self.assertEqual( s2["n"]["p"]["value"].getValue(), plug["value"].getValue() )
+		self.assertEqual( s2["n"]["p"]["value"].defaultValue(), plug["value"].defaultValue() )
+		self.assertEqual( s2["n"]["p"]["name"].getValue(), plug["name"].getValue() )
+		self.assertEqual( s2["n"]["p"]["name"].defaultValue(), plug["name"].defaultValue() )
+
+		if "enable" in plug.keys():
```

Comment:
"enable" -> "enabled", and the same below. Also `"enabled" in plug` is sufficient - no need to call `keys()`.